net/tsdial/tsdial.go

```diff
@@ -129,6 +129,20 @@ func (d *Dialer) SetExitDNSDoH(doh string) {
 	}
 }
 
+// SetRoutes configures the dialer to dial the specified routes via Tailscale,
+// and the specified localRoutes using the default interface.
+func (d *Dialer) SetRoutes(routes, localRoutes []netip.Prefix) {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+	d.routes = bart.Table[bool]{}
```

Comment:
you could make the table first, and then acquire the mutex once you're ready to assign to d.routes.

Reply:
I've been considering that. But `bart.Table` contains a `sync.Once`, and I believe it should not be copied after its first used by any operation on the table.

Reply:
you should be storing it as a pointer, not a value.

Reply:
(note that all its methods are on the pointer receiver... and see https://go.dev/wiki/CodeReviewComments#receiver-type)

Reply:
Done.